solver/jobs.go

```diff
@@ -178,6 +180,7 @@ func (s *state) setEdge(index Index, targetEdge *edge, targetState *state) {
 			targetState.mpw.Add(s.mpw)
 			targetState.allPw[s.mpw] = struct{}{}
 		}
+		targetState.owners = append(targetState.owners, s.vtx)
```

Comment:
Do we need this extra property? Can't we just cycle the `.owner` chains of all sibling edges?

Reply:
Yup, looks like that works as well :tada: